esp-hal/src/lib.rs

```diff
@@ -182,9 +182,22 @@
 // MUST be the first module
 mod fmt;
 
+#[macro_use]
+/// This module contains code generated by `esp-metadata`.
```

Comment:
Do we even want to publicly document this? Maybe one (and probably should have some mention here anyways) for the DEVELOPER-GUIDELINES?

Reply:
This macro shouldn't be publicly visible at all, the documentation is meant to be for us. D-G could use a section about how a driver looks, in general.